src/codegen/generators/expression_generator.rs

```diff
@@ -182,6 +190,20 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
         }
     }
 
+    fn register_debug_location(&self, statement: &AstStatement) -> Result<(), Diagnostic> {
+        let function_context = self
+            .function_context
+            .expect("Cannot generate debug info without function context");
+        let line = function_context
+            .new_lines
+            .get_line_nr(statement.get_location().get_start());
+        let column = function_context
+            .new_lines
+            .get_column(line, statement.get_location().get_start());
+        self.debug
```

Comment:
I find this .set_debug_location a bit surprising - I was wondering what you are setting in the DebugBuilder.
To be honest I would find `self.llvm.set_current_debug_location(self.debug.create_debug_location(function, line, colum), context)` clearer.

Reply:
There were two things i'm trying to avoid here, first introducing the DITypes and DILocation into codegen instead of keeping them to the debug module, which admittedly is not important, and second I don't want to generate debug info unless there is something to generate, so now llvm would also suddenly need to test for debug levels.
We could talk about this next week, maybe there's a clearer way to do this i wouldn't mind.